### PR TITLE
Updates Pages Projects with more stuff

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
## Description

Adds field, `production_branch`, to pages project and deployments.

## Has your change been tested?

Updated unit tests and they pass

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [ ] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs. soon:tm:

### CODE REVIEW

File: pages_project.go [hunk 1]
```diff
@@ -15,11 +15,12 @@ type PagesProject struct {
 	CreatedOn           *time.Time                    `json:"created_on"`
 	SubDomain           string                        `json:"subdomain"`
 	Domains             []string                      `json:"domains,omitempty"`
-	Source              PagesProjectSource            `json:"source"`
+	Source              PagesProjectSource            `json:"source,omitempty"`
```
Comment (WalshyDev) [inline pages_project.go]:
```suggestion
	Source              *PagesProjectSource           `json:"source,omitempty"`
```

File: pages_project.go [hunk 2]
```diff
@@ -39,11 +40,11 @@ type PagesProjectSourceConfig struct {
...
-	BuildCommand      string `json:"build_command"`
-	DestinationDir    string `json:"destination_dir"`
-	RootDir           string `json:"root_dir"`
-	WebAnalyticsTag   string `json:"web_analytics_tag"`
-	WebAnalyticsToken string `json:"web_analytics_token"`
+	BuildCommand      string `json:"build_command,omitempty"`
+	DestinationDir    string `json:"destination_dir,omitempty"`
+	RootDir           string `json:"root_dir,omitempty"`
+	WebAnalyticsTag   string `json:"web_analytics_tag,omitempty"`
+	WebAnalyticsToken string `json:"web_analytics_token,omitempty"`
```
Comment (WalshyDev) [inline pages_project.go]:
```suggestion
	BuildCommand      string `json:"build_command"`
	DestinationDir    string `json:"destination_dir"`
	RootDir           string `json:"root_dir"`
	WebAnalyticsTag   string `json:"web_analytics_tag"`
	WebAnalyticsToken string `json:"web_analytics_token"`
```

**Comment by WalshyDev** [comment]
Big API doc PR is up 😆 

Let's make sure everything is synced 👍

**Comment by Cyb3r-Jak3** [comment]
Pretty sure I got everything that is being added to the docs. Will have @WalshyDev verify

File: pages_project.go [hunk 3]
```diff
@@ -30,11 +31,15 @@ type PagesProjectSource struct {
...
-	ProductionBranch   string `json:"production_branch"`
-	PRCommentsEnabled  bool   `json:"pr_comments_enabled"`
-	DeploymentsEnabled bool   `json:"deployments_enabled"`
+	Owner                        string   `json:"owner"`
+	RepoName                     string   `json:"repo_name"`
+	ProductionBranch             string   `json:"production_branch"`
+	PRCommentsEnabled            bool     `json:"pr_comments_enabled"`
+	DeploymentsEnabled           bool     `json:"deployments_enabled"`
+	ProductionDeploymentsEnabled bool     `json:"production_deployments_enabled"`
+	PreviewDeploymentSetting     string   `json:"preview_deployment_setting"`
```
Comment (WalshyDev) [inline pages_project.go]: This is an enum, can we have a proper type for it rather than allowing any value

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: pages_project.go [hunk 4]
```diff
@@ -117,6 +127,23 @@ type pagesProjectListResponse struct {
...
+}
+
+type R2BindingValue struct {
+	Name string `json:"name"`
+}
+type R2BindingMap map[string]*R2BindingValue
+
+type D1BindingMap map[string]*D1Binding
+
+type D1Binding struct {
```
Comment (WalshyDev) [inline pages_project.go]:
```suggestion
type R2BindingMap map[string]*R2BindingValue

type R2BindingValue struct {
	Name string `json:"name"`
}

type D1BindingMap map[string]*D1BindingValue

type D1BindingValue struct {
```

File: pages_project.go [hunk 5]
```diff
@@ -30,11 +31,15 @@ type PagesProjectSource struct {
...
+	Owner                        string   `json:"owner"`
+	RepoName                     string   `json:"repo_name"`
+	ProductionBranch             string   `json:"production_branch"`
+	PRCommentsEnabled            bool     `json:"pr_comments_enabled"`
+	DeploymentsEnabled           bool     `json:"deployments_enabled"`
+	ProductionDeploymentsEnabled bool     `json:"production_deployments_enabled"`
+	PreviewDeploymentSetting     string   `json:"preview_deployment_setting"`
+	PreviewBranchIncludes        []string `json:"preview_branch_includes"`
+	PreviewBranchExcludes        []string `json:"preview_branch_excludes"`
 }
```
Comment (WalshyDev) [inline pages_project.go:52]:
```suggestion

type PreviewDeploymentSetting string

const (
	AllBranches    PreviewDeploymentSetting = "all"
	NoBranches     PreviewDeploymentSetting = "none"
	CustomBranches PreviewDeploymentSetting = "custom"
)

```

File: pages_project.go (hunk 3 above)
Comment (WalshyDev) [reply pages_project.go]:
```suggestion
	PreviewDeploymentSetting     PreviewDeploymentSetting `json:"preview_deployment_setting"`
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Comment by WalshyDev** [comment]
I believe these are all the properties indeed

**Review by WalshyDev — APPROVED** [review]
lgtm

**Comment by jacobbednarz** [comment]
LGTM. tracking this one against the internal PRs and lines up from what i can see, thanks!

**Pull request merged**